x/subscription/keeper/subscription.go

```diff
@@ -256,20 +188,91 @@ func (k Keeper) CreateSubscription(
 		)
 	}
 
-	k.SetSubscription(ctx, sub)
+	if !found {
+		expiry := NextMonth(ctx.BlockTime()).UTC().Unix()
+		k.subsTS.AddTimerByBlockTime(ctx, uint64(expiry), []byte(consumer), []byte{})
+		err = k.subsFS.AppendEntry(ctx, consumer, block, &sub)
+	} else {
+		k.subsFS.ModifyEntry(ctx, consumer, sub.Block, &sub)
+	}
 
-	return nil
+	return err
+}
+
+func (k Keeper) advanceMonth(ctx sdk.Context, subkey []byte) {
+	date := ctx.BlockTime()
+	block := uint64(ctx.BlockHeight())
+	consumer := string(subkey)
+
+	var sub types.Subscription
+	_ = k.subsFS.FindEntry(ctx, consumer, block, &sub)
```

Comment:
handle the return value

Reply:
Right, I meant to put a panic here. We come here only from the timer so the entry must exist, and if it doesn't we really should not to continue, as something bad happened internally.

Reply:
fixed: 7ee029b6